packages/core/src/api/Editor.ts

```diff
@@ -0,0 +1,433 @@
+import { Editor as TiptapEditor } from "@tiptap/core";
+import {
+  DOMParser,
+  DOMSerializer,
+  Node as PMNode,
+  Node,
+  Schema,
+} from "prosemirror-model";
+import { unified } from "unified";
+import remarkParse from "remark-parse";
+import remarkStringify from "remark-stringify";
+import remarkGfm from "remark-gfm";
+import remarkRehype from "remark-rehype";
+import rehypeParse from "rehype-parse";
+import rehypeStringify from "rehype-stringify";
+import rehypeRemark from "rehype-remark";
+import { getBlockInfoFromPos } from "../extensions/Blocks/helpers/getBlockInfoFromPos";
+import { Style, StyledText } from "../extensions/Blocks/api/styleTypes";
+import {
+  Block,
+  BlockProps,
+  blockProps,
+  PartialBlock,
+} from "../extensions/Blocks/api/blockTypes";
+import { TextCursorPosition } from "../extensions/Blocks/api/cursorPositionTypes";
+import { simplifyBlocks } from "./simplifyBlocksRehypePlugin";
+import { removeUnderlines } from "./removeUnderlinesRehypePlugin";
+
+export function blockToNode(block: PartialBlock, schema: Schema) {
+  let content: PMNode[] = [];
+
+  if (typeof block.content === "string") {
+    content.push(schema.text(block.content));
+  } else if (typeof block.content === "object") {
+    for (const styledText of block.content) {
+      const marks = [];
+
+      for (const style of styledText.styles) {
```

Comment:
did we decide on richtext vs styledtext?

Reply:
Don't think we made the final call on that yet, should probably jump on a quick call for that.